components/StoryCard.tsx

```diff
@@ -62,11 +62,17 @@ const StoryCard = ({ story, state, index, addFilter }: StoryCardParams): JSX.Ele
   const { id }: UrlParams = router.query;
 
   const [isModalVisible, setIsModalVisible] = useState(false);
-  const [name, setName] = useState(story.name);
+  const [name, setName] = useState<string>(story.name);
+  let escape = false; // we can't use setState for this as the dispatch of this takes an extra tick.
```

Comment:
(nit) I think `useRef` may be a good option for this, so the variable doesn't get redefine on each render. Not a big deal tho since the value is just tweaked a millisecond after being used. But just wanted to point it. 🙂